homer0/projext
# feat(project): add support for .env files

### What does this PR do?

> If you don't know what an `.env` file is, please read about [`dotenv`](https://yarnpkg.com/en/package/dotenv).

It adds support for `.env` files that can be scoped to your targets. The files can load environment variables that projext will send to the Node runner and the browser targets configuration and the "definitions" build engines replace on the bundles.

You can read more about the feature on the updated `projectConfiguration.md`.

#### why is this PR breaking?

First, `BuildNodeRunner` and `BuildTraspiler` now require `Utils`: until now, they were using it from inside the `Targets` injection, with a `@todo` saying that I would eventually change it... well, I did it.

And `Targets` now requires `dotEnvUtils`, which is a new service that takes care of loading and parsing the `.env` files.

### How should it be tested manually?

As for now, this can only be tested on Node targets that don't require bundling nor transpilation; The build engines will later add the rest of the support.

Now, create a file called `.env.[target-name]` and declare some variable, then, on your target code, log that variable.

When you run your target, you should have access to the variable as it was declared on the `.env` file.

And of course...

```bash
yarn test
# or
npm test
```
